### PR TITLE
Fix clash bug

### PR DESCRIPTION
**Pull request by wish2023**
Updated code to check for clash at night time. More bugs may come once #191 is fixed as I can't test too much without program crashing.

### CODE REVIEW

File: src/main/java/athena/TaskList.java [hunk 1]
```diff
@@ -175,8 +174,12 @@ private boolean isTimeClash(Task taskToCompare, Task task) {
 
     private boolean isIndividualTimeClash(LocalTime taskStartTime, LocalTime taskEndTime,
                                           LocalTime existingTaskStartTime, LocalTime existingTaskEndTime) {
-        return !(taskEndTime.compareTo(existingTaskStartTime) <= 0
-                || taskStartTime.compareTo(existingTaskEndTime) >= 0);
+        boolean isTimeClash = !(taskEndTime.compareTo(existingTaskStartTime) <= 0
+                || (taskStartTime.compareTo(existingTaskEndTime) >= 0
+                && existingTaskEndTime.compareTo(existingTaskStartTime) > 0));
+        boolean isMidnightClash = taskEndTime.compareTo(taskStartTime) < 0
+                && existingTaskEndTime.compareTo(existingTaskStartTime) < 0;
```
Comment (daniellimws) [inline src/main/java/athena/TaskList.java:181]: Do we need to check the existing tasks?

**Review by daniellimws — APPROVED** [review]
Looks good! Let's merge this asap

File: src/main/java/athena/TaskList.java (hunk 1 above)
Comment (wish2023) [reply src/main/java/athena/TaskList.java:181]: Yup for `isMidnightClash` the algo basically checks if both the end times are less than their respective start times. If so, both tasks go past 12am which means there's a clash at the 12am mark.

Comment (daniellimws) [reply src/main/java/athena/TaskList.java:181]: Shouldn't existing tasks not go past the 12am mark? Wouldn't we have stopped that from happening when it was added/edited?

Comment (wish2023) [reply src/main/java/athena/TaskList.java:181]: As in ATHENA still alows task to be added provided they end at 12am but not past that

**Pull request merged**